packages/devtools_server/lib/src/server.dart

```diff
@@ -233,3 +269,20 @@ void printOutput(
 }) {
   print(machineMode ? jsonEncode(json) : message);
 }
+
+/// Map the URI (which may already be Observatory web app) to a WebSocket URI
+/// for the VM service.
+///
+/// If the URI is already a VM Service WebSocket URI it will not be modified.
+Uri getVmServiceUriFromObservatoryUri(Uri uri) {
```

Comment:
Does the naming of this method still make sense? Seems odd to me that we have observatory in the name.

Reply:
Ah, that is a good point.

Reply:
I wasn't sure what to call it. The banner says "Observatory listening on" and then gives a HTTP URL. The service URI we want is a websocket protocol and has `/ws` on the end. What should we call each of them?

Reply:
`getVmServiceUriFromObservatoryUri()` ==> `convertToWebsocketUrl()`? Or, 

```
String convertToWebsocketUrl({@required String serviceProtocolUrl}) {
  ...
}
```

Reply:
That works for me - though I think making it named is a bit overkill, as it's never likely to be a literal string, so we'll always end up with `convertToWebSocketUrl(serviceProtocolUrl: serviceProtocolUrl)`.

Also, since the types are Uris, we should probably use Uri?

```dart
Uri convertToWebsocketUri(Uri serviceProtocolUri) {
  ...
}
```

I still think "serviceProtocolUri" to mean the non-websocket part is weird, but I don't the the specific name is that important as long as it conveys roughly what it does - the doc comments can always be more detailed if required.

@devoncarew I haven't made any code use this yet, so changing it and re-publishing is fine by me. Since I can't merge publish, feel free to just change it if it's easier than waiting for me.